datastore/importers/nmdb-import-systeminfo/nmdb-import-systeminfo.cpp

```diff
@@ -0,0 +1,121 @@
+
+#include <netmeld/datastore/tools/AbstractImportSpiritTool.hpp>
+#include "Parser.hpp"
+
+namespace nmdt = netmeld::datastore::tools;
+
+// =============================================================================
+// Import tool definition
+// =============================================================================
+template<typename P, typename R>
+class Tool : public nmdt::AbstractImportSpiritTool<P,R>
+{
+  // ===========================================================================
+  // Variables
+  // ===========================================================================
+  private:
+  protected:
+  public:
+
+  // ===========================================================================
+  // Constructors
+  // ===========================================================================
+  private:
+  protected:
+  public:
+    Tool() : nmdt::AbstractImportSpiritTool<P,R>
+      (
+       "systeminfo",           // command line tool imports data from
+       PROGRAM_NAME,           // program name (set in CMakeLists.txt)
+       PROGRAM_VERSION         // program version (set in CMakeLists.txt)
+      )
+    {}
+
+  // ===========================================================================
+  // Methods
+  // ===========================================================================
+  private: // Methods part of internal API
+    void
+    specificInserts(pqxx::transaction_base& t) override
+    {
+      const auto& toolRunId {this->getToolRunId()};
+      const auto& deviceId  {this->getDeviceId()};
+      for (auto& results : this->tResults) {
+
+        LOG_DEBUG << "Iterating over DeviceInformation\n";
+        results.devInfo.save(t, toolRunId, deviceId);
+        LOG_DEBUG << results.devInfo.toDebugString() << "\n";
+
+        LOG_DEBUG << "Iterating over OperatingSystems\n";
+        results.os.save(t, toolRunId, deviceId);
+        LOG_DEBUG << results.os.toDebugString() << "\n";
+
+        LOG_DEBUG << "Iterating over Hotfixes\n";
+        t.exec_prepared("insert_raw_hotfixes", toolRunId, ConvertVectorToPostgresArray(results.hotfixes));
+        for (auto& hotfix : results.hotfixes) {
+          LOG_DEBUG << hotfix << "\n";
+        }
+
+        LOG_DEBUG << "Iterating over Interfaces\n";
+        for (auto& [_, result] : results.network_cards)
+        {
+          result.save(t, toolRunId, deviceId);
+          LOG_DEBUG << result.toDebugString() << "\n";
+        }
+
+        LOG_DEBUG << "Iterating over Observations\n";
+        results.observations.save(t, toolRunId, deviceId);
+        LOG_DEBUG << results.observations.toDebugString() << "\n";
+      }
+    }
+    std::string
+    ConvertVectorToPostgresArray(const std::vector<std::string>& vec) {
+      std::string result = "{";
+
+      for (const auto& element : vec) {
+          result += "'" + element + "',";
+      }
+
+      if (!vec.empty()) {
+          result.pop_back();
+      }
+
+      result += "}";
+      return result;
+    }
```

Comment:
Again, do not do this.
1. The logic is wrong.  You are adding single quotes and, thus, changing the hotfix value.  So any search is now `'''KBXXXXXX'''` instead of just `'KBXXXXXX'`.  As I mentioned before you are working with libpqxx not a Postgresql command directly.
2. I just tested it, nothing is needed besides just inserting the vector.  The libpqxx calls handle everything for you.

So this whole function can go away.

Reply:
Okay removed